src/types.ts

```diff
@@ -1,17 +1,27 @@
 export interface EmbeddedEdgeConfig {
   digest: string;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  items: Record<string, any>;
+  items: Record<string, EdgeConfigValue>;
 }
 
 /**
  * Edge Config Client
  */
 export interface EdgeConfigClient {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  get: <T = any>(key: string) => Promise<T | undefined>;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  getAll: <T = any>(keys?: (keyof T)[]) => Promise<T | undefined>;
+  get: <T extends EdgeConfigValue = EdgeConfigValue>(
+    key: string,
+  ) => Promise<T | undefined>;
+  getAll: <T extends EdgeConfigItems = EdgeConfigItems>(
+    keys?: (keyof T)[],
+  ) => Promise<T>;
   has: (key: string) => Promise<boolean>;
   digest: () => Promise<string>;
 }
+
+export type EdgeConfigItems = Record<string, EdgeConfigValue>;
+export type EdgeConfigValue =
+  | string
+  | number
+  | boolean
+  | null
+  | { [x: string]: EdgeConfigValue }
```

Comment:
should maybe be a Record<string, EdgeConfigValue>?

Reply:
I'm not a typescript wizard by any means but it seems like it can't be one?

<img width="646" alt="image" src="https://user-images.githubusercontent.com/1765075/217221147-522097fa-03fa-4633-870f-bbf29f080b3d.png">


https://www.typescriptlang.org/play?#code/C4TwDgpgBAogJgcwgYQPYDsBmBLBA1AQwBsBXCAQSgF4AoKKAHygGdgAnbdBOxqdEgLYAjCGx5MhqVEQgF04viSJEFAbygBtAB4AuFu04IAunvhI0WXIVIUoAXxqOnT0JFiIUGHPmJkAQtQKrBxcCvzCogqS0rLy9Ez8ygoAShAAxqhscAA8wYYANO7mXla+EH4AfEA

Reply:
Just realized the changes I made prevent interfaces to be passed as the generic now

<img width="1222" alt="image" src="https://user-images.githubusercontent.com/1765075/217475033-bcc7fb30-4930-437e-b4a4-4e86f99292a5.png">


Sample code

```ts
type EdgeConfigType = {
  a: string;
  b: number;
};


interface EdgeConfigInterface {
  a: string;
  b: number;
};

async function main() {
  const _allT = await getAll<EdgeConfigType>();
  const _selectedT = await getAll<EdgeConfigType>(["a", "b"]);

  const _allI = await getAll<EdgeConfigInterface>();
  const _selectedI = await getAll<EdgeConfigInterface>(["a", "b"]);

  console.log(_allT, _selectedT);
  console.log(_allI, _selectedI);
}
```

cc @jzxhuang